binr/r2r/run.c

```diff
@@ -1203,6 +1203,19 @@ R_API R2RTestResultInfo *r2r_run_test(R2RRunConfig *config, R2RTest *test) {
 	}
 	}
 	bool broken = r2r_test_broken (test);
+#if ASAN
+# if !R2_ASSERT_STDOUT
+# error R2_ASSERT_STDOUT undefined or 0
+# endif
+	R2RProcessOutput *out = ret->proc_out;
+	if (!success && test->type == R2R_TEST_TYPE_CMD && strstr (test->path, "/dbg")
+	    && (!out->out ||
+	        (!strstr (out->out, "WARNING:") && !strstr (out->out, "ERROR:") && !strstr (out->out, "FATAL:")))
+	    && (!out->err ||
+	        (!strstr (out->err, "Sanitizer") && !strstr (out->err, "runtime error:")))) {
+		broken = true;
```

Comment:
why set to broken and not to fail?

Reply:
if it's set to fail, the build is not going to be green

Reply:
maybe i'm missing something...
What is this check about? What kind of tests should fall into this `if`?

Reply:
Last time when I used Ubuntu 20.04 instead of the Travis Docker image, the output of some of the debug tests changed. I suspect that the differences are due to the tests being dependent on shared libraries from the environment, which can change depending on the distribution used.

Unfortunately, libasan6 [is only available on Ubuntu 20.04 and above](https://packages.ubuntu.com/search?suite=all&section=all&arch=any&keywords=libasan6&searchon=names) so I don't think there's much option regarding the Ubuntu version.

Only 7 tests should be affected by this.

Reply:
So by doing this we do expect some tests to fail, right?

Reply:
See https://github.com/radareorg/radare2/pull/17594/checks?check_run_id=1096771366#step:13:49 (starting from line 49 of the "Run tests" section) for the failing debug tests.

Reply:
If `ubuntu-latest` == `ubuntu-20.04` then I think this check can and should be dropped.